qwerty_mlir/lib/Qwerty/IR/QwertyOps.cpp

```diff
@@ -252,44 +254,47 @@ struct CallIndirectConst : public mlir::OpRewritePattern<qwerty::CallIndirectOp>
     }
 };
 
+bool block_is_duplicatable(mlir::Block &in_block) {
+    for (mlir::Operation &op : in_block) {
+        if (!mlir::isPure(&op)) {
+            return false;
+        }
+        for (mlir::Value operand : op.getOperands()) {
+            if (llvm::isa<qcirc::NonStationaryTypeInterface>(operand.getType()) && operand.getParentBlock() != &in_block) {
+                return false;
+            }
+        }
+    }
+    return true;
+};
+
 struct CallIndirectIf : public mlir::OpRewritePattern<qwerty::CallIndirectOp> {
+
     using OpRewritePattern<qwerty::CallIndirectOp>::OpRewritePattern;
 
     mlir::LogicalResult matchAndRewrite(qwerty::CallIndirectOp call,
                                         mlir::PatternRewriter &rewriter) const override {
         mlir::scf::IfOp if_op =
             call.getCallee().getDefiningOp<mlir::scf::IfOp>();
-        if (!if_op
-                // If there are many results used all over the place, we can't
-                // safely repurpose it as spitting out the results of the call
-                || if_op.getResults().size() != 1
-                // If there is someone else using the function this is
-                // returning, we can't safely get rid of this scf.if
-                || !if_op.getResults()[0].hasOneUse()) {
+        if (!if_op || if_op.getResults().size() != 1) {
```

Comment:
Could you leave a `TODO` comment here suggesting that we should generalize this to IfOps that yield more than 1 value? (Sometimes I search the codebase for `TODO` to look for things to fix)